### PR TITLE
feat: Return group info on stage commit

### PR DESCRIPTION
**Pull request by augustocdias**

### CODE REVIEW

File: openmls/src/group/mls_group/membership.rs [hunk 1]
```diff
@@ -21,14 +23,14 @@ impl MlsGroup {
     /// update of the committer's leaf [KeyPackage].
     ///
     /// If successful, it returns a tuple of [`MlsMessageOut`]s, where the first
```
Comment (beltram) [inline openmls/src/group/mls_group/membership.rs:25]:
```suggestion
    /// If successful, it returns a triple of [`MlsMessageOut`]s, where the first
```

File: openmls/src/group/mls_group/updates.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ impl MlsGroup {
     /// If not, a new one will be created on the fly.
     ///
     /// If successful, it returns a tuple of [`MlsMessageOut`] (containing the
```
Comment (beltram) [inline openmls/src/group/mls_group/updates.rs:14]:
```suggestion
    /// If successful, it returns a triple of [`MlsMessageOut`] (containing the
```

**Review by beltram — CHANGES_REQUESTED** [review]

File: openmls/tests/test_external_commit.rs [hunk 3]
```diff
@@ -120,3 +120,108 @@ fn test_external_commit(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoPr
...
+        ProcessedMessageContent::StagedCommitMessage(commit) => {
+            alice_group.merge_staged_commit(*commit);
+        }
+        _ => panic!("Unexpected message type"),
+    }
+
+    // bob sends a message to alice
+    let message = bob_group.create_message(backend, b"Hello Alice").unwrap();
+
+    let msg_in = MlsMessageIn::try_from_bytes(&message.to_bytes().unwrap()).unwrap();
```
Comment (beltram) [inline openmls/tests/test_external_commit.rs:220]: can't it just be `message.into()` ?

**Review by beltram — CHANGES_REQUESTED** [review]

File: openmls/tests/test_external_commit.rs [hunk 4]
```diff
@@ -120,3 +120,108 @@ fn test_external_commit(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoPr
...
+        MlsGroup::new(
+            backend,
+            &group_config,
+            alice_cb.credential().signature_key(),
+        )
+        .expect("An unexpected error occurred.")
+    };
+
+    // Self update Alice's to get a group info from a commit
+    let (_, _, group_info) = alice_group.self_update(backend, None).unwrap();
```
[the remark this answers is not in the document]
Comment (augustocdias) [reply openmls/tests/test_external_commit.rs:165]: That is not valid. We would have to pattern match it or make a if let expression:

```
error[E0005]: refutable pattern in local binding: `(_, _, None)` not covered
   --> openmls/tests/test_external_commit.rs:165:9
    |
165 |     let (.., Some(group_info)) = alice_group.self_update(backend, None).unwrap();
    |         ^^^^^^^^^^^^^^^^^^^^^^ pattern `(_, _, None)` not covered
    |
    = note: `let` bindings require an "irrefutable pattern", like a `struct` or an `enum` with only one variant
    = note: for more information, visit https://doc.rust-lang.org/book/ch18-02-refutability.html
    = note: the matched value is of type `(openmls::prelude_test::MlsMessageOut, Option<openmls::prelude_test::MlsMessageOut>, Option<openmls::messages::GroupInfo>)`
```

**Review by beltram — APPROVED** [review]

**Pull request closed without merging**